kube/cron/hof_db_table_replacer.yaml

```diff
@@ -0,0 +1,103 @@
+          - name: hof-db-table-replacer
+            image: quay.io/ukhomeofficedigital/hof-db-table-replacer@sha256:241e0d20f87f4dbeb988a026d0d84590edf65a4a3b8a1f46dd67592f919297b9
+            imagePullPolicy: Always
+            securityContext:
+              runAsNonRoot: true
+            envFrom:
+              - configMapRef:
+                  {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+                  name: configmap-{{ .DRONE_SOURCE_BRANCH }}
+                  {{ else }}
+                  name: configmap
+                  {{ end }}
+            env:
+              - name: TZ
+                value: Europe/London
+              - name: NODE_TLS_REJECT_UNAUTHORIZED
+                value: "0"
+              - name: NOTIFY_KEY
+                valueFrom:
+                  secretKeyRef:
+                    name: notify-key
+                    key: notify-key
+              - name: FILE_VAULT_URL
+              {{ if eq .KUBE_NAMESPACE .PROD_ENV }}
+                value: fv-ima.sas.homeoffice.gov.uk/file
+#              {{ else if eq .KUBE_NAMESPACE .STG_ENV }}
+#                value: fv-ima.stg.sas.homeoffice.gov.uk/file
+              {{ else if eq .KUBE_NAMESPACE .UAT_ENV }}
+                value: fv-ima.uat.sas-notprod.homeoffice.gov.uk/file
+              {{ else if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+                value: https://fv-{{ .DRONE_SOURCE_BRANCH }}.branch.sas-notprod.homeoffice.gov.uk/file
+              {{ end }}
+              - name: KEYCLOAK_SECRET
+                valueFrom:
+                  secretKeyRef:
+                    name: keycloak-client
+                    key: secret
+              - name: KEYCLOAK_CLIENT_ID
+                valueFrom:
+                  secretKeyRef:
+                    name: keycloak-client
+                    key: id
+              - name: KEYCLOAK_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    name: keycloak-user
+                    key: username
+              - name: KEYCLOAK_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: keycloak-user
+                    key: password
+              - name: DB_REPLACER_TARGET_TABLE
+                valueFrom:
+                  configMapKeyRef:
+                    {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+                    name: configmap-{{ .DRONE_SOURCE_BRANCH }}
+                    {{ else }}
+                    name: configmap
+                    {{ end }}
+                    key: DB_REPLACER_TARGET_TABLE
+              - name: DB_REPLACER_SOURCE_FILE_TABLE
+                valueFrom:
+                  configMapKeyRef:
+                    {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+                    name: configmap-{{ .DRONE_SOURCE_BRANCH }}
+                    {{ else }}
+                    name: configmap
+                    {{ end }}
+                    key: DB_REPLACER_SOURCE_FILE_TABLE
+              
```

Comment:
These two variables will already be in the environment from the config map (lines 29-35). The ones we need to add to this manifest are:

```bash
- name: DB_HOST
    valueFrom:
      secretKeyRef:
        key: endpoint
        {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
        name: ima-notprod-rds
        {{ else }}
        name: ima-prod
        {{ end }}
  - name: DB_NAME
    valueFrom:
      secretKeyRef:
        key: db_name
        {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
        name: ima-notprod-rds
        {{ else }}
        name: ima-prod
        {{ end }}
  - name: DB_USER
    valueFrom:
      secretKeyRef:
        key: username
        {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
        name: ima-notprod-rds
        {{ else }}
        name: ima-prod
        {{ end }}
  - name: DB_PASS
    valueFrom:
      secretKeyRef:
        key: password
        {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
        name: ima-notprod-rds
        {{ else }}
        name: ima-prod
        {{ end }}
  - name: DB_PORT
    value: "5432"
  - name: DB_INSERT_BATCH_SIZE
    value: "2000"
  - name: DB_CLIENT
    value: knex
  - name: DB_MODEL
    value: knex-postgres-model
  - name: SERVICE_NAME
    value: ima
```

Reply:
Hi @jamiecarterHO , Thanks for providing me these information. I have updated it now.